File: tests/fixtures/typescript/sample.ts
// A sample TypeScript module for the extraction tests.

export interface Widget {
  id: number;
  title: string;
}

export type WidgetMap = Record<string, Widget>;

enum Status {
  Draft,
  Published,
}

export namespace Registry {
  export const LIMIT = 100;

  export function register(widget: Widget): void {
    console.log(widget);
  }

  export namespace Cache {
    export function clear(): void {}
  }
}

declare function inspect(widget: Widget): string;

export abstract class Repository {
  abstract find(id: number): Widget;

  count(): number {
    return 0;
  }
}

export class WidgetService {
  private widgets: Widget[] = [];

  add(widget: Widget): void {
    this.widgets.push(widget);
  }

  get size(): number {
    return this.widgets.length;
  }
}

export const MAX_WIDGETS = 25;

export const makeWidget = (title: string): Widget => ({
  id: 1,
  title,
});

File: tests/fixtures/python/sample.py
"""A sample Python module for the extraction tests."""

import functools

VERSION = "1.2.3"
MAX_RETRIES: int = 3
_cache = {}


def plain(value):
    """Return the value untouched."""
    return value


async def fetch(url):
    """Pretend to fetch the URL."""
    return url


@functools.cache
def memoized(value):
    """Return the value, cached."""
    return value


def outer():
    """Hold a nested helper."""

    def inner():
        return 1

    return inner


class Widget:
    """A widget."""

    @property
    def title(self):
        return self._title

    class Meta:
        """Nested widget metadata."""

        def describe(self):
            return "meta"


class Registry(Widget):
    """A registry of widgets."""

    def register(self, widget):
        self.widgets.append(widget)
